test/cpp/codegen/codegen_test_minimal.cc

```diff
@@ -16,9 +16,14 @@
  *
  */
 
+#include <grpcpp/impl/codegen/completion_queue.h>
 #include <gtest/gtest.h>
 
 namespace grpc {
+
+// Unused implementation for the virtual "Next" method.
+bool CompletionQueue::Next(void** tag, bool* ok) { return false; }
```

Comment:
@dgquintas : does this feel kosher to you? We need a mechanism to allow the use of codegen headers without crapping out on linker errors now that we're using `virtual` members for some classes that get used in codegen. Can you comment on this?

Reply:
including codegen headers around is never a problem. The problem is putting non-codegen things into the codegen hierarchy. This should be fine AFAICT.